pwa-devdocs/src/technologies/basic-concepts/internationalization/index.md

```diff
@@ -0,0 +1,150 @@
+
+## Translation dictionaries
+
+Translation dictionary files contain key/value pairs for localized text.
+PWA Studio's i18n framework uses a similar dictionary approach for translation files as Magento, but
+instead of a CSV format, it uses JSON.
+
+Dictionary files must be inside an `i18n` directory and use their target locale for their filename.
+For example: `en_US.json`, `en_GB.json`, `fr_FR.json`.
+
+Example `en_US.json` content:
+
+```json
+{
+    "About Us": "About Us",
+    "Hi, {name}": "Hi, {name}",
+    "accountMenu.orderHistoryLink": "Order History",
+    "autocomplete.resultSummary": "{resultCount} items"
+}
+```
+
+### ID formats
+
+The JSON object's keys act as unique IDs for localized text.
+The example posted shows the different ways you can identify the localized text in a dictionary file.
+
+The first and second entry uses the original `en_US` locale text as the ID for the translated text.
+This is the same approach used in Magento.
+
+The third and fourth entry uses an alternative, dot notation approach for the defining the ID for translated text.
+This helps to identify the component and element where the text appears.
+
+Both approach have their pros and cons, and developers are free to choose which approach works for them when they develop their own components and storefronts.
+
+### Message syntax
+
+The i18n framework accepts the same [message syntax][] as the underlying `react-intl` library.
+Along with static text, this syntax supports variables, dates, and even conditional formatting.
+
+To translate text with variables, pass in a mapping object to the `values` prop in the `FormattedMessage` component.
+
+{% raw %}
+
+```jsx
+return(
+    <FormattedMessage
+    id="Hi, {name}"
+    defaultMessage="Hi, {name}"
+    values={{
+        name: 'Veronica',
+    }}
+    />
+)
+```
+
+{% endraw %}
+
+When using the `formatMessage()` function, pass in the mapping object as the second parameter.
+
+```jsx
+const text = formatMessage(
+    { id:"Hi, {name}", defaultMessage="Hi, {name}"},
+    { name: currentUser.firstname }
+);
+```
+
+For more details, see the [message syntax][] documentation at FormatJS.
+
+## Language packages and plugins
+
+Language packages provide translation data for one or more locales.
+They are also used to override the text in the same locale.
+
+Unlike the core Magento application, which install language packages through Composer, PWA Studio storefronts install language packages as NPM dependencies.
+
+An NPM dependency is a language package if it meets the following criteria:
+
+- The package contains an intercept file
+- The intercept file sets the special feature `i18n` flag to `true` for the package
+- The package contains an `i18n` directory
+- The `i18n` directory contains a dictionary file with a locale formatted name
+
+<!-- TODO: Create an in-depth tutorial for creating a language package extension -->
```

Comment:
Isn't there an example repo @davemacaulay created for this? Not sure if it's public but would be trivial to make one and just link it, I'm sure.

Reply:
Yes we have https://github.com/magento-research/pwa-studio-target-experiments/tree/2939ebf9af871f8ad43739554ad91e31afc49963/packages/i18n-package and a PR to get it merged here: https://github.com/magento-research/pwa-studio-target-experiments/pull/2

Reply:
since it's not merged into mainline yet, I can't add a link to it. maybe when I do the tutorial I can add it.